src/Text/Pandoc/Readers/Vimwiki.hs

```diff
@@ -0,0 +1,661 @@
+code = try $ do
+  char '`'
+  contents <- many1Till (noneOf "\n") (char '`')
+  return $ B.code contents
+
+superscript :: PandocMonad m => VwParser m Inlines
+superscript = try $
+  B.superscript <$> mconcat <$> (char '^' >> many1Till inline' (char '^'))
+
+subscript :: PandocMonad m => VwParser m Inlines
+subscript = try $
+  B.subscript <$> mconcat <$> (string ",," 
+    >> many1Till inline' (try $ string ",,"))
+
+link :: PandocMonad m => VwParser m Inlines
+link = try $ do 
+  string "[["
+  contents <- lookAhead $ manyTill anyChar (string "]]")
+  case '|' `elem` contents of 
+                  False -> do
+                    manyTill anyChar (string "]]") 
+-- not using try here because [[hell]o]] is not rendered as a link in vimwiki
+                    return $ B.link (procLink contents) "" (B.str contents)
+                  True  -> do 
+                    url <- manyTill anyChar $ char '|'
+                    lab <- mconcat <$> (manyTill inline $ string "]]")
+                    return $ B.link (procLink url) "" lab
+
+image :: PandocMonad m => VwParser m Inlines
+image = try $ do 
+  string "{{"
+  contentText <- lookAhead $ manyTill (noneOf "\n") (try $ string "}}")
+  images $ length $ filter (== '|') contentText
+
+images :: PandocMonad m => Int -> VwParser m Inlines
+images k
+  | k == 0 = do 
+           imgurl <- manyTill anyChar (try $ string "}}")
+           return $ B.image (procImgurl imgurl) "" (B.str "")
+  | k == 1 = do
+           imgurl <- manyTill anyChar (char '|')
+           alt <- mconcat <$> (manyTill inline $ (try $ string "}}"))
+           return $ B.image (procImgurl imgurl) "" alt
+  | k == 2 = do
+           imgurl <- manyTill anyChar (char '|')
+           alt <- mconcat <$> (manyTill inline $ char '|')
+           attrText <- manyTill anyChar (try $ string "}}")
+           return $ B.imageWith (makeAttr attrText) (procImgurl imgurl) "" alt
+  | otherwise = do
+           imgurl <- manyTill anyChar (char '|')
+           alt <- mconcat <$> (manyTill inline $ char '|')
+           attrText <- manyTill anyChar (char '|')
+           manyTill anyChar (try $ string "}}")
+           return $ B.imageWith (makeAttr attrText) (procImgurl imgurl) "" alt
+
+procLink' :: String -> String
+procLink' s
+  | ((take 6 s) == "local:") = "file" ++ (drop 5 s) 
+  | ((take 6 s) == "diary:") = "diary/" ++ (drop 6 s) ++ ".html"
+  | or ((`isPrefixOf` s) <$> [ "http:", "https:", "ftp:", "file:", "mailto:", 
+                              "news:", "telnet:" ]) 
+                             = s
+  | s == ""                  = ""
+  | (last s) == '/'          = s
+  | otherwise                = s ++ ".html"
+  
+procLink :: String -> String
+procLink s = let (x, y) = divByHash "" s in (procLink' x) ++ y
+
+divByHash :: String -> String -> (String, String)
+divByHash s1 s2
+  | s2 == ""        = (s1, s2)
+  | head s2 == '#'  = (s1, s2) 
+  | otherwise       = divByHash (s1 ++ [head s2]) (tail s2)
+
```

Comment:
If the point of this function is to split a string on a prefix not containing `#` and a suffix starting with `#`, why not use the Prelude function `break`?

```haskell
procLink s = procLink' x ++ y
  where (x, y) = break (=='#') s
```